taca/element/Element_Runs.py

```diff
@@ -0,0 +1,1270 @@
+        grouped_df = df_samples.groupby(
+            [
+                "I1Mask",
+                "I2Mask",
+                "I1UmiMask",
+                "I2UmiMask",
+                "R1Mask",
+                "R2Mask",
+                "settings",
+            ]
+        )
+
+        # Sanity check
+        if sum([len(group) for _, group in grouped_df]) < len(df_samples):
+            msg = "Some samples were not included in any submanifest."
+            logging.error(msg)
+            raise AssertionError(msg)
+        elif sum([len(group) for _, group in grouped_df]) > len(df_samples):
+            logging.warning("Some samples were included in multiple submanifests.")
+
+        # Iterate over groups to build composite manifests
+        manifest_root_name = f"{self.NGI_run_id}_demux"
+        manifests = []
+        n = 0
+        for (
+            I1Mask,
+            I2Mask,
+            I1UmiMask,
+            I2UmiMask,
+            R1Mask,
+            R2Mask,
+            settings,
+        ), group in grouped_df:
+            file_name = f"{manifest_root_name}_{n}.csv"
+
+            runValues_section = "\n".join(
+                [
+                    "[RUNVALUES]",
+                    "KeyName, Value",
+                    f"manifest_file, {file_name}",
+                    f"manifest_group, {n+1}/{len(grouped_df)}",
+                    f"built_from, {manifest_to_split}",
+                ]
+            )
+
+            # Instantiate settings
+            settings_kvs = {
+                "R1FastqMask": R1Mask,
+                "I1Mask": I1Mask,
+                "I2Mask": I2Mask,
+                "R2FastqMask": R2Mask,
+            }
+
+            # Add UMI settings
+            if "Y" in I1UmiMask and "Y" not in I2UmiMask:
+                settings_kvs["UmiMask"] = I1UmiMask
+                settings_kvs["UmiFastQ"] = "TRUE"
+            elif "Y" in I2UmiMask and "Y" not in I1UmiMask:
+                settings_kvs["UmiMask"] = I2UmiMask
+                settings_kvs["UmiFastQ"] = "TRUE"
+            elif "Y" not in I1UmiMask and "Y" not in I2UmiMask:
+                pass
+            else:
+                raise AssertionError("Both I1 and I2 appear to contain UMIs.")
+
+            # Unpack settings from LIMS manifest
+            if settings:
+                for kv in settings.split(" "):
+                    k, v = kv.split(":")
+                    settings_kvs[k] = v
+
+            settings_section = "\n".join(
+                [
+                    "[SETTINGS]",
+                    "SettingName, Value",
+                ]
+                + [f"{k}, {v}" for k, v in settings_kvs.items()]
+            )
+
+            # Add PhiX stratified by index length
+            group_controls = df_controls[
+                df_controls["Lane"].isin(group["Lane"].unique())
+            ].copy()
+
+            # Trim PhiX indexes to match group
+            i1_len = group["Index1"].apply(len).max()
```

Comment:
I must say I don't have the full picture, but would there be any risk for index collision introduced here when the phix indices are trimmed, @kedhammar?

Reply:
The index collision check done by LIMS only take into account the index lengths prior to trimming. So I guess *in theory* it's possible that the trimmed PhiX index might correspond to a full length sample index.

Do you want me to add an assertion for it?